src/main/java/com/example/accommodationbookingapp/controller/PaymentController.java

```diff
@@ -0,0 +1,71 @@
+package com.example.accommodationbookingapp.controller;
+
+import com.example.accommodationbookingapp.dto.payment.CreatePaymentDto;
+import com.example.accommodationbookingapp.dto.payment.PaymentResponseDto;
+import com.example.accommodationbookingapp.model.Payment;
+import com.example.accommodationbookingapp.service.payment.PaymentService;
+import com.example.accommodationbookingapp.service.url.UriService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/payments")
+@RequiredArgsConstructor
+@Tag(name = "Payment management", description = "Endpoints for managing payments")
+public class PaymentController {
+    private PaymentService paymentService;
+    private UriService uriService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping("/{id}")
+    @Operation(summary = "Get Payments by User ID",
+            description = "Get Payments as dto by User ID")
+    public List<PaymentResponseDto> getPaymentsByUserId(@PathVariable Long id) {
+        return paymentService.findAllByUserId(id);
+    }
```

Comment:
I don't know if I understand this correctly but wouldn't it be better to send user id based on Authentication principal?